Add CI, pre-commit hooks

brucellino opened the pull request:
This PR adds a few bits and pieces to help maintaining the project:

1. Ansible role dependencies are explicitly declared (#268)
2. Add pre-commit hooks to ensure that clean code is committed (can be extended later)
3. Add a basic CI task (#267)


Closed #268 
Closed #267
Closed #270

NicolasLiampotis committed:
Merge pull request #262 from rciam/devel

Fix support for SSPv2

dependabot[bot] committed:
Bump ansible from 2.10.7 to 8.5.0

Bumps [ansible](https://github.com/ansible-community/ansible-build-data) from 2.10.7 to 8.5.0.
- [Changelog](https://github.com/ansible-community/ansible-build-data/blob/main/docs/release-process.md)
- [Commits](https://github.com/ansible-community/ansible-build-data/compare/2.10.7...8.5.0)

---
updated-dependencies:
- dependency-name: ansible
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

brucellino committed:
chore: add pre-commit hooks

Signed-off-by: Bruce Becker <brucellino@protonmail.ch>

brucellino committed:
chore: add ansible-lint pre-commit hook

Signed-off-by: Bruce Becker <brucellino@protonmail.ch>

brucellino committed:
build(deps): add ansible role declarations

WIP #268

Signed-off-by: Bruce Becker <brucellino@protonmail.ch>

brucellino committed:
chore: only run ansible-lint pre-commit on playbooks

Signed-off-by: Bruce Becker <brucellino@protonmail.ch>

brucellino committed:
ci: add ansible lint task

Signed-off-by: Bruce Becker <brucellino@protonmail.ch>

brucellino committed:
Merge pull request #2 from brucellino/ci/validate

Ci/validate

brucellino committed:
Merge pull request #1 from brucellino/dependabot/pip/ansible-8.5.0

Bump ansible from 2.10.7 to 8.5.0

brucellino committed:
chore(deps): update ansible ecosystem

Signed-off-by: Bruce Becker <brucellino@protonmail.ch>

brucellino committed:
style(apiserver): update playbook to pass lint

Signed-off-by: Bruce Becker <brucellino@protonmail.ch>

brucellino committed:
Merge pull request #3 from brucellino/fix/apiserver

Fix/apiserver

NicolasLiampotis commented on requirements.txt:
```diff
@@ -1,4 +1,7 @@
-ansible==2.10.7
+ansible==9.3.0
+ansible-base==2.10.17
+ansible-core==2.16.4
```
Some roles are not ready for ansible-core > v2.14:

```
ERROR! [DEPRECATED]: ansible.builtin.include has been removed. Use include_tasks or import_tasks instead. This feature was removed from ansible-core in a release after 2023-05-16. Please update your playbooks.
```

While ansible-core v2.14 requires ansible v7, it seems that this setup works fine without needing ansible-base v2.10

```
ansible==7.7.0
ansible-core==2.14.14
```

brucellino replied:
Ok, I was aware of that and wanted to do a Mmmmmmega PR! with all of the roles updated, but wanted to keep this one small. Quite happy to update all of the playbooks and roles, it won't change functionality. What do you think?

NicolasLiampotis replied:
Sounds good! How about we proceed with merging this PR, which bumps the ansible/ansible-core versions in a backwards compatible way and adds the lint workflow for the playbooks?

Then, you can submit a separate Mmmmmmega PR specifically for updating all the playbooks and roles for the latest & greatest ansible/ansible-core.

brucellino replied:
I think a better approach might be to tag a release to the HEAD just before we merge this, and then declare a breaking change when we merge?

brucellino replied:
Done in bc6cbeb

NicolasLiampotis commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,31 @@
+      files: ^.*.yml
+      exclude: .github
+      additional_dependencies:
+        - ansible-core>=2.16.0
```
See comment about ansible-core version

brucellino replied:
Trying to find a version for ansible-lint which will make the existing roles pass a check is more difficult than I expected. I suggest just accepting that the roles will not pass lint now, letting the pipeline fail, and slowly converge in the Big PR.

NicolasLiampotis reviewed (commented):
🎉 Thanks a bunch for the PR! 🙏

Could you check our my comments on the branches and the ansible-core vs ansible-base vs ansible versions?

brucellino commented on .github/workflows/lint.yml:
```diff
@@ -0,0 +1,13 @@
+# .github/workflows/ansible-lint.yml
+name: ansible-lint
+on:
+  pull_request:
+    branches: ["main"]
```
Done in f97e0b2

brucellino commented:
@NicolasLiampotis from my side, ready to merge.

brucellino committed:
style: Conform to Ansible > 2.10 (#4)

style(attrauth): fix lint issues in playbook
style(apache): fix ansible-lint issues in defaults and handlers
style(apache): fix lint in OS-dependent configuration tasks
style(apache): fix linting issues in ssl config tasks
style(apache): apply ansible-lint's opinionated fix to role
style(apache): align name of handler to respect ansible-lint rules
style(apache): apply lint fixes to main tasks
style(comange-registry): apply linting rules to defaults
style(comange-registry): apply linting rules to configure tasks
style(comanage): declare no changes in ad-hoc commands
  This should be updated when we know how to identify a change in order to ensure idempotency
style(comanage): split long install command over several lines
style(comanage): apply linting fixes to install tasks
style(comanage): explicitly ignore idempotency in cache clear task
style(comanage): explicitly ignore idempotency in static files tasks
style(php): fix linting errors in role defaults
style(php): fix linting errors in role defaults
style(php): fix linting errors in role taks and vars
style(shibboleth-sp): fix linting errors in role defaults
style(shibboleth-sp): fix linting errors in role main tasks
style(shibboleth-sp): fix all linting errors in role
WIP #270
WIP #267
style(common): apply linting rules to role defaults
style(common): apply linting rules to role defaults
style(common): fix linting errors in main role tasks
style(common): apply linting fixes to role tasks and vars
style(ssp): make role defaults pass ansible lint
chore(deps): freeze current requirements for dependabot
style(ssp): apply style rules to ssp handlers
style(ssp): reduce linting warnings in role composer tasks
style(ssp): apply linting rules to configure-common tasks
style(ssp): ensure configure-saml and shib tasks pass lint
style(ssp): ensure authservers playbook passes Ansible lint
WIP #270
style(authzkeys): conform to Ansible Lint style
style(cache): conform to ansible style guide
style(cert2saml): conform to ansible style for ssp module role
style(comanage): conform comanage playbook and roles to Ansible style guide
style(d4science): conform playbook to Ansible lint
style(postgres): conform db playbook and postgres role to ansible lint
style(egi-igtf): conform EGI IGTF role and playbook to Ansible Lint
refactor(egi-igtf): move task to handler
style(deep2saml): conform playbook to Ansible Lint
style(federation-registry): conform to ansible lint
Some task modules have been changed from postgresql_query to postgresql_script
style(fedreg): conform to ansible lint
style(firewall): conform to ansible lint
style(google2saml): conform to ansible lint
chore: ignore editor cruft
style(keycloak): conform to Ansible Lint
style(keycloak): conform to Ansible Lint
style(metrics): conform metrics servers to Ansible Lint
style(monservers): conform monservers to Ansible Lint
style(oauth2saml): conform to Ansible Lint
style(ldap): conform to Ansible Lint
style(orcid): conform to Ansible Lint
style(rciam-keycloak,nginx,rsyslog,webproxy): conform to Ansible Lint
style(oidc): conform to Ansible Lint
style(registry-ui): conform to Ansible Lit
chore: add detect-secrets pre-commit hook
style(saml,oidc): conform to Ansible Lint
style(secrets): conform to Ansible Lint
style(site): conform to Ansible Lint
style(ssp): conform to Ansible Lint
fix(apache): fix module name
style(utils): conform to Ansible Lint
style: conform to Ansible Lint

---------

Signed-off-by: Bruce Becker <brucellino@protonmail.ch>
Signed-off-by: Bruce Becker <bruce.becker@egi.eu>

brucellino commented:
@NicolasLiampotis This now passes Ansible Lint
I had to refactor a few of the tasks in order to make it so, but no functionality has been changed.

brucellino commented:
bump @NicolasLiampotis
Can we discuss whether we can merge this?

NicolasLiampotis commented:
> bump @NicolasLiampotis Can we discuss whether we can merge this?

Before merging this, I suggest closing #272 and #274 to ensure we have a stable release in master. Merging now could potentially break several roles in the devel branch.

brucellino commented:
Ok, no problem. Do you guys have a test case described somewhere for #272 and #274 ? Can I help create it if not?

dependabot[bot] committed:
Bump dnspython from 2.1.0 to 2.6.1 (#6)

Bumps [dnspython](https://github.com/rthalley/dnspython) from 2.1.0 to 2.6.1.
- [Release notes](https://github.com/rthalley/dnspython/releases)
- [Changelog](https://github.com/rthalley/dnspython/blob/main/doc/whatsnew.rst)
- [Commits](https://github.com/rthalley/dnspython/compare/v2.1.0...v2.6.1)

---
updated-dependencies:
- dependency-name: dnspython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump jinja2 from 3.1.3 to 3.1.4 (#7)

Bumps [jinja2](https://github.com/pallets/jinja) from 3.1.3 to 3.1.4.
- [Release notes](https://github.com/pallets/jinja/releases)
- [Changelog](https://github.com/pallets/jinja/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/jinja/compare/3.1.3...3.1.4)

---
updated-dependencies:
- dependency-name: jinja2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cryptography from 42.0.5 to 43.0.1 (#8)

Bumps [cryptography](https://github.com/pyca/cryptography) from 42.0.5 to 43.0.1.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/42.0.5...43.0.1)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansible-core from 2.16.4 to 2.17.6 (#9)

Bumps [ansible-core](https://github.com/ansible/ansible) from 2.16.4 to 2.17.6.
- [Release notes](https://github.com/ansible/ansible/releases)
- [Commits](https://github.com/ansible/ansible/compare/v2.16.4...v2.17.6)

---
updated-dependencies:
- dependency-name: ansible-core
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansible-core from 2.17.6 to 2.17.7 (#10)

Bumps [ansible-core](https://github.com/ansible/ansible) from 2.17.6 to 2.17.7.
- [Release notes](https://github.com/ansible/ansible/releases)
- [Commits](https://github.com/ansible/ansible/compare/v2.17.6...v2.17.7)

---
updated-dependencies:
- dependency-name: ansible-core
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cryptography from 43.0.1 to 44.0.1 (#11)

Bumps [cryptography](https://github.com/pyca/cryptography) from 43.0.1 to 44.0.1.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/43.0.1...44.0.1)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump jinja2 from 3.1.4 to 3.1.5 (#12)

Bumps [jinja2](https://github.com/pallets/jinja) from 3.1.4 to 3.1.5.
- [Release notes](https://github.com/pallets/jinja/releases)
- [Changelog](https://github.com/pallets/jinja/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/jinja/compare/3.1.4...3.1.5)

---
updated-dependencies:
- dependency-name: jinja2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump jinja2 from 3.1.5 to 3.1.6 (#13)

Bumps [jinja2](https://github.com/pallets/jinja) from 3.1.5 to 3.1.6.
- [Release notes](https://github.com/pallets/jinja/releases)
- [Changelog](https://github.com/pallets/jinja/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/jinja/compare/3.1.5...3.1.6)

---
updated-dependencies:
- dependency-name: jinja2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansible from 9.3.0 to 12.2.0 (#15)

Bumps [ansible](https://github.com/ansible-community/ansible-build-data) from 9.3.0 to 12.2.0.
- [Changelog](https://github.com/ansible-community/ansible-build-data/blob/main/docs/release-process.md)
- [Commits](https://github.com/ansible-community/ansible-build-data/compare/9.3.0...12.2.0)

---
updated-dependencies:
- dependency-name: ansible
  dependency-version: 12.2.0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump filelock from 3.13.1 to 3.20.1 (#16)

Bumps [filelock](https://github.com/tox-dev/py-filelock) from 3.13.1 to 3.20.1.
- [Release notes](https://github.com/tox-dev/py-filelock/releases)
- [Changelog](https://github.com/tox-dev/filelock/blob/main/docs/changelog.rst)
- [Commits](https://github.com/tox-dev/py-filelock/compare/3.13.1...3.20.1)

---
updated-dependencies:
- dependency-name: filelock
  dependency-version: 3.20.1
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump filelock from 3.20.1 to 3.20.3 (#17)

Bumps [filelock](https://github.com/tox-dev/py-filelock) from 3.20.1 to 3.20.3.
- [Release notes](https://github.com/tox-dev/py-filelock/releases)
- [Changelog](https://github.com/tox-dev/filelock/blob/main/docs/changelog.rst)
- [Commits](https://github.com/tox-dev/py-filelock/compare/3.20.1...3.20.3)

---
updated-dependencies:
- dependency-name: filelock
  dependency-version: 3.20.3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cryptography from 44.0.1 to 46.0.5 (#18)

Bumps [cryptography](https://github.com/pyca/cryptography) from 44.0.1 to 46.0.5.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/44.0.1...46.0.5)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-version: 46.0.5
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump pygments from 2.17.2 to 2.20.0 (#21)

Bumps [pygments](https://github.com/pygments/pygments) from 2.17.2 to 2.20.0.
- [Release notes](https://github.com/pygments/pygments/releases)
- [Changelog](https://github.com/pygments/pygments/blob/master/CHANGES)
- [Commits](https://github.com/pygments/pygments/compare/2.17.2...2.20.0)

---
updated-dependencies:
- dependency-name: pygments
  dependency-version: 2.20.0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cryptography from 46.0.5 to 46.0.6 (#20)

Bumps [cryptography](https://github.com/pyca/cryptography) from 46.0.5 to 46.0.6.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/46.0.5...46.0.6)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-version: 46.0.6
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cryptography from 46.0.6 to 46.0.7 (#22)

Bumps [cryptography](https://github.com/pyca/cryptography) from 46.0.6 to 46.0.7.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/46.0.6...46.0.7)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-version: 46.0.7
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cryptography from 46.0.7 to 48.0.1 (#23)

Bumps [cryptography](https://github.com/pyca/cryptography) from 46.0.7 to 48.0.1.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/46.0.7...48.0.1)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-version: 48.0.1
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ansible-core from 2.17.7 to 2.18.18rc1 (#24)

Bumps [ansible-core](https://github.com/ansible/ansible) from 2.17.7 to 2.18.18rc1.
- [Release notes](https://github.com/ansible/ansible/releases)
- [Commits](https://github.com/ansible/ansible/compare/v2.17.7...v2.18.18rc1)

---
updated-dependencies:
- dependency-name: ansible-core
  dependency-version: 2.18.18rc1
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cryptography from 48.0.1 to 50.0.0 (#25)

Bumps [cryptography](https://github.com/pyca/cryptography) from 48.0.1 to 50.0.0.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/48.0.1...50.0.0)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-version: 50.0.0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
